src/prefect/tasks/spacy/spacy_tasks.py

```diff
@@ -0,0 +1,216 @@
+from prefect import Task
+from prefect.utilities.tasks import defaults_from_attrs
+
+import spacy
+
+
+class SpacyNLP(Task):
+    """
+    Task for processing text with a spaCy pipeline.
+
+    Args:
+        - text (unicode, optional): string to be processed, can be provided during construction
+            or when task is run
+        - nlp (spaCy text processing pipeline, optional): a custom spaCy text
+            processing pipeline, if provided, this pipeline will be used instead
+            of being created from spacy_model_name
+        - spacy_model_name (str, optional): name of the spaCy language model, default
+            model is 'en_core_web_sm', will be ignored if nlp is provided
+        - disable (List[str], optional): list of pipeline components
+            to disable, only applicable to pipelines loaded from spacy_model_name
+        - component_cfg (dict, optional): a dictionary with extra keyword
+            arguments for specific components, only applicable to pipelines loaded from
+            spacy_model_name
+        - **kwargs (dict, optional): additional keyword arguments to pass to the
+            Task constructor
+    """
+
+    def __init__(
+        self,
+        text: str = u"",
+        nlp=None,
+        spacy_model_name: str = "en_core_web_sm",
+        disable: list = [],
+        component_cfg: dict = {},
+        **kwargs
+    ):
+        self.text = text
+
+        ## load spacy model
+        if nlp:
+            self.nlp = nlp
+        else:
+            try:
+                self.nlp = spacy.load(
+                    spacy_model_name, disable=disable, component_cfg=component_cfg
+                )
+            except IOError:
+                raise ValueError("spaCy model %s not found." % spacy_model_name)
+
+        super().__init__(**kwargs)
+
+    @defaults_from_attrs("text")
+    def run(self, text: str = u""):
```

Comment:
Any chance you could write a test to verify that `defaults_from_attrs` works the way we expect in this situation?  Specifically, a test which verifies that if a non-trivial `text` was provided at initialization and then nothing is provided at runtime, that the initialized `text` attribute it used?  You could just initialize the class with a `MagicMock` `nlp` attribute and verify that it was called with the correct `text`.

Reply:
Added test in latest commits